mm2src/coins/lightning/ln_rpc.rs

```diff
@@ -1,73 +1,181 @@
-use crate::utxo::rpc_clients::{electrum_script_hash, ElectrumClient, UtxoRpcClientOps, UtxoRpcError};
+use super::*;
+#[cfg(not(target_arch = "wasm32"))]
+use crate::utxo::rpc_clients::UtxoRpcClientEnum;
+use crate::utxo::rpc_clients::{BlockHashOrHeight, EstimateFeeMethod};
+use crate::utxo::utxo_standard::UtxoStandardCoin;
+use crate::{MarketCoinOps, MmCoin};
+#[cfg(not(target_arch = "wasm32"))]
+use bitcoin::blockdata::block::BlockHeader;
 use bitcoin::blockdata::script::Script;
 use bitcoin::blockdata::transaction::Transaction;
 use bitcoin::consensus::encode;
 use bitcoin::hash_types::Txid;
-use common::block_on;
-use common::mm_error::prelude::MapToMmFutureExt;
+use bitcoin_hashes::Hash;
+use common::executor::spawn;
+use common::{block_on, log};
 use futures::compat::Future01CompatExt;
+use keys::hash::H256;
 use lightning::chain::{chaininterface::{BroadcasterInterface, ConfirmationTarget, FeeEstimator},
                        Filter, WatchedOutput};
-use rpc::v1::types::Bytes as BytesJson;
+use rpc::v1::types::H256 as H256Json;
+use std::convert::TryFrom;
 
-impl FeeEstimator for ElectrumClient {
+impl FeeEstimator for UtxoStandardCoin {
     // Gets estimated satoshis of fee required per 1000 Weight-Units.
-    // TODO: use fn estimate_fee instead of fixed number when starting work on opening channels
     fn get_est_sat_per_1000_weight(&self, confirmation_target: ConfirmationTarget) -> u32 {
-        match confirmation_target {
+        let conf = &self.as_ref().conf;
+        // TODO: Maybe default_fee and confirmation targets can be set in coin configs or lightning configs (would require to move lightning config to coin config) instead
+        let default_fee = match confirmation_target {
             // fetch background feerate
             ConfirmationTarget::Background => 253,
             // fetch normal feerate (~6 blocks)
             ConfirmationTarget::Normal => 2000,
             // fetch high priority feerate
             ConfirmationTarget::HighPriority => 5000,
-        }
+        } * 4;
```

Comment:
Could you please clarify the meaning of this multiplication by 4?

Reply:
The function returns the required fee in sats per 1000 Weight-Units, the transaction weight is equal to the transaction actual size divided by 4 because of the block increase post-Segwit. Since the lowest fee acceptable per 1000 weights is 253 as described here https://github.com/lightningdevkit/rust-lightning/blob/9fcc626ee4d23a276fc8dd87ddb2538f4d5565f9/lightning/src/chain/chaininterface.rs#L41-L51, the default fee per KB will be 253 * 4 for background confirmation and 2000 * 4 for normal, etc...

Reply:
I see, thanks for the clarification!